File: official/docs/java/batches/LabelBatch.java
package batches;

import com.easypost.EasyPost;
import com.easypost.exception.EasyPostException;
import com.easypost.model.Batch;

import java.util.HashMap;

public class LabelBatch {
    public static void main(String[] args) throws EasyPostException {
        EasyPost.apiKey = System.getenv("EASYPOST_API_KEY");

        Batch batch = Batch.retrieve("batch_...");

        HashMap<String, Object> labelMap = new HashMap<String, Object>();
        labelMap.put("file_format", "PDF");

        batch.label(labelMap);

        System.out.println(batch);
    }
}

File: official/docs/csharp/batches/LabelBatch.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Xunit;
using Xunit.Sdk;
using Newtonsoft.Json;
using EasyPost;

namespace EasyPostExamples;

public class Examples
{
    [Fact]
    public async Task LabelBatch()
    {
        string apiKey = Environment.GetEnvironmentVariable("EASYPOST_API_KEY")!;

        EasyPost.ClientManager.SetCurrent(apiKey);

        Batch batch = await Batch.Retrieve("batch_...");

        await batch.GenerateLabel("PDF");

        new TestOutputHelper().WriteLine(JsonConvert.SerializeObject(batch, Formatting.Indented));
    }
}
